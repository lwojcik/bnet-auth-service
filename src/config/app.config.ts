import { registerAs } from '@nestjs/config';
import { trueStringToBoolean } from '../utils/trueStringToBoolean';
import { APP, APP_INFO, DEFAULTS } from '../common/constants';

const { env } = process;
const defaultValue = DEFAULTS.app;

export const appConfig = registerAs('app', () => ({
  name: APP_INFO.name,
  environment: env[APP.environment],
  host: env[APP.host] || defaultValue.host,
  port: parseInt(env[APP.port], 10) || defaultValue.port,
  enableCors:
    trueStringToBoolean({ value: env[APP.enableCors] }) ||
    defaultValue.enableCors,
  corsOrigin: env[APP.corsOrigin] || defaultValue.corsOrigin,
}));
