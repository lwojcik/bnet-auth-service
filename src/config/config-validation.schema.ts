import * as Joi from 'joi';
import { CronExpression } from '@nestjs/schedule';
import { validateRegionName } from '../common/validators/validateRegionName.validator';

const appSchema = {
  NODE_ENV: Joi.string().default('production'),
  BAS_APP_HOST: Joi.string().default('0.0.0.0'),
  BAS_APP_PORT: Joi.string().default(3000),
  BAS_APP_CORS_ORIGIN: Joi.optional(),
};

const redisSchema = {
  BAS_REDIS_HOST: Joi.string().optional().default('redis'),
  BAS_REDIS_PORT: Joi.string().optional().default('6379'),
  BAS_REDIS_PASSWORD: Joi.any().optional(),
  BAS_REDIS_TTL_SECS: Joi.string().optional().default('2000'),
  BAS_REDIS_DB: Joi.string().optional().default('0'),
};

const battlenetSchema = {
  BAS_BATTLENET_REGION: Joi.string().required().custom(validateRegionName),
  BAS_BATTLENET_CLIENT_ID: Joi.string().required(),
  BAS_BATTLENET_CLIENT_SECRET: Joi.string().required(),
};

const throttleSchema = {
  BAS_THROTTLE_LIMIT: Joi.string().optional().default('300'),
  BAS_THROTTLE_TTL_SECS: Joi.string().optional().default('60'),
};

const authSchema = {
  BAS_AUTH_USERNAME: Joi.string().optional(),
  BAS_AUTH_JWT_SECRET: Joi.string().optional(),
};

const httpsSchema = {
  BAS_HTTPS_KEY_PATH: Joi.string().optional(),
  BAS_HTTPS_CERT_PATH: Joi.string().optional(),
};

const cronSchema = {
  BAS_CRON_PATTERN: Joi.string()
    .optional()
    .default(CronExpression.EVERY_30_MINUTES),
};

export const configValidationSchema = Joi.object({
  ...appSchema,
  ...redisSchema,
  ...battlenetSchema,
  ...throttleSchema,
  ...authSchema,
  ...httpsSchema,
  ...cronSchema,
});
