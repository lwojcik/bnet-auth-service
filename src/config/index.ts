export * from './app.config';
export * from './auth.config';
export * from './battlenet.config';
export * from './endpoints.config';
export * from './redis.config';
export * from './throttle.config';
