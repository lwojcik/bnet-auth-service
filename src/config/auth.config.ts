import { registerAs } from '@nestjs/config';
import { trueStringToBoolean } from '../utils/trueStringToBoolean';
import { AUTH, DEFAULTS } from '../common/constants';

const { env } = process;
const defaultValue = DEFAULTS.auth;

export const authConfig = registerAs('auth', () => ({
  enable:
    trueStringToBoolean({ value: process.env[AUTH.enable] }) ||
    defaultValue.enable,
  username: env[AUTH.username] || defaultValue.username,
  jwtSecret: env[AUTH.jwtSecret] || defaultValue.jwtSecret,
}));
