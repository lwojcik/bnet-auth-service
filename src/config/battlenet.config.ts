import { registerAs } from '@nestjs/config';
import { RegionName } from 'blizzapi';

export const battleNetConfig = registerAs('battlenet', () => ({
  region: process.env.BAS_BATTLENET_REGION as RegionName,
  clientId: process.env.BAS_BATTLENET_CLIENT_ID,
  clientSecret: process.env.BAS_BATTLENET_CLIENT_SECRET,
}));
