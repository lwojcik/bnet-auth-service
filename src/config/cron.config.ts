import { registerAs } from '@nestjs/config';

export const cronConfig = registerAs('cron', () => ({
  enable: process.env.BAS_CRON_ENABLE === 'true',
  pattern: process.env.BAS_CRON_PATTERN,
}));
