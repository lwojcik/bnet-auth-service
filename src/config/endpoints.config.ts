import { registerAs } from '@nestjs/config';

export const endpointsConfig = registerAs('endpoints', () => ({
  status: {
    url: '/status',
    method: 'GET',
  },
  accesstoken: {
    url: '/accesstoken',
    method: 'GET',
  },
  accesstokenrefresh: {
    url: '/accesstoken?refresh=true',
    method: 'GET',
  },
}));
