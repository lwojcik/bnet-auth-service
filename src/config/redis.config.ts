import { registerAs } from '@nestjs/config';
import { trueStringToBoolean } from '../utils/trueStringToBoolean';
import { REDIS, DEFAULTS } from '../common/constants';

const { env } = process;
const defaultValue = DEFAULTS.redis;

export const redisConfig = registerAs('redis', () => ({
  enable: trueStringToBoolean({ value: env[REDIS.enable] }),
  host: env[REDIS.host] || defaultValue.host,
  port: parseInt(env[REDIS.port], 10) || defaultValue.port,
  password: env[REDIS.password] || defaultValue.password,
  ttlSecs: parseInt(env[REDIS.ttlSecs], 10) || defaultValue.ttlSecs,
  db: parseInt(env[REDIS.db], 10) || defaultValue.db,
  keyName: env[REDIS.keyName] || defaultValue.keyName,
  keyPrefix: env[REDIS.keyPrefix] || defaultValue.keyPrefix,
}));
