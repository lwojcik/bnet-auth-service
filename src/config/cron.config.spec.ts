import { cronConfig } from './cron.config';

describe('cronConfig', () => {
  it('should create cron config', () => {
    expect(cronConfig()).toMatchSnapshot();
  });
});
