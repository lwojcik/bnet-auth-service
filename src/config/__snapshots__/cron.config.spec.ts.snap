// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cronConfig should create cron config 1`] = `
Object {
  "enable": false,
  "pattern": undefined,
}
`;
